src/librustc_typeck/check/mod.rs

```diff
@@ -939,24 +949,33 @@ impl<'a, 'gcx, 'tcx> Visitor<'gcx> for GatherLocalsVisitor<'a, 'gcx, 'tcx> {
 
     // Add explicitly-declared locals.
     fn visit_local(&mut self, local: &'gcx hir::Local) {
-        let o_ty = match local.ty {
+        let local_ty = match local.ty {
             Some(ref ty) => {
                 let o_ty = self.fcx.to_ty(&ty);
+                debug!("visit_local: ty.hir_id={:?} o_ty={:?} c_ty={:?}", ty.hir_id, o_ty, 1);
+
+                let revealed_ty = if self.fcx.tcx.features().impl_trait_in_bindings {
+                    self.fcx.require_type_is_sized(o_ty, ty.span, traits::SizedReturnType);
```

Comment:
Is this line needed/right, @oli-obk? If so, is `traits::SizedReturnType` the right value of argument to pass?

Reply:
CC @cramertj @eddyb

Reply:
@cramertj Advice on this, specifically the value of the third arg?

Reply:
This looks correct to me.